pkg/webhook/resourcedistribution/validating/utils.go

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validating
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+)
+
+// GetTargetNamespaces will return the intersection of them when multiple options are enabled.
+func GetTargetNamespaces(handlerClient client.Client, targets *appsv1alpha1.ResourceDistributionTargets, fldPath *field.Path) (selectedNamespaces []string, allErrs field.ErrorList) {
+	threshold := 0
+	selectedTimes := make(map[string]int)
```

Comment:
can we make it a sets of strings, does we really care about the counts?

Reply:
ResourceDistribution.Spec.Targets provides 4 ways(options) for users to select target namespaces. 
If more than one options were enabled, this function will return the intersection of their results.  (For example, users  can use this rule to exclude some namespaces.)
Thus, this function has to record the counts in order to calculate this intersection.

Reply:
i still think counts is not necessary, consider:
1. first put all namespaces selected by includedNamespaces and NamespaceLabelSelector into the selectedTimes sets
2. then remove all namespaces selected by excludedNamespaces from selectedTimes the sets

Reply:
If so, we need add another field(e.g., "AllNamespaces") to ensure users have a way to distribute  resource to all namespaces